physionet-django/project/authorization/access.py

```diff
@@ -0,0 +1,123 @@
+
+    credentialed_with_dua_signed = Q(
+        access_policy=AccessPolicy.CREDENTIALED
+    ) & Q(duasignature__in=dua_signatures)
+
+    query = required_training_complete & (
+        contributor_review_with_access | credentialed_with_dua_signed
+    )
+    return query
+
+
+def get_projects_accessible_through_events(user):
+    events_all = Event.objects.filter(Q(host=user) | Q(participants__user=user))
+    active_events = set(events_all.filter(end_date__gte=datetime.now()))
+    accessible_datasets = EventDataset.objects.filter(event__in=active_events, is_active=True)
+    accessible_projects_ids = []
+    for event_dataset in accessible_datasets:
+        if event_dataset.has_access(user):
+            accessible_projects_ids.append(event_dataset.dataset.id)
+    query = Q(id__in=accessible_projects_ids)
+    return query
+
+
+def get_accessible_projects(user):
+    """
+    Returns query filter for published projects accessible by a specified user
+    """
+    query = Q(deprecated_files=False)
+
+    query &= get_public_projects_query()
+
+    if user.is_authenticated:
+        query |= get_restricted_projects_query(user)
+
+    if user.is_credentialed:
+        query |= get_credentialed_projects_query(user)
+
+    query |= get_projects_accessible_through_events(user)
+
+    return PublishedProject.objects.filter(query).distinct()
+
+
+def can_access_project(project, user):
+    """Checks if the project is accessible by the user"""
+    if project.deprecated_files:
+        return False
+
+    if not project.allow_file_downloads:
+        return False
+
+    if project.access_policy == AccessPolicy.OPEN:
+        return True
+    elif project.access_policy == AccessPolicy.RESTRICTED:
+        return user.is_authenticated and DUASignature.objects.filter(project=project, user=user).exists()
+    elif project.access_policy == AccessPolicy.CREDENTIALED:
+        return (
+            user.is_authenticated
+            and user.is_credentialed
+            and DUASignature.objects.filter(project=project, user=user).exists()
+            and Training.objects.get_valid()
+            .filter(training_type__in=project.required_trainings.all(), user=user)
+            .count()
+            == project.required_trainings.count()
+        )
+    elif project.access_policy == AccessPolicy.CONTRIBUTOR_REVIEW:
+        return (
+            user.is_authenticated
+            and user.is_credentialed
+            and DataAccessRequest.objects.get_active(
+                project=project,
+                requester=user,
+                status=DataAccessRequest.ACCEPT_REQUEST_VALUE
+            ).exists()
+            and Training.objects.get_valid()
+            .filter(training_type__in=project.required_trainings.all(), user=user)
+            .count()
+            == project.required_trainings.count()
+        )
+    return False
+
+
+def can_view_project_files(project, user):
+    """
+    Checks if the project files are accessible by the user
```

Comment:
It would be good to use the docstrings as an opportunity to clarify the difference between `can_view_project_files` and `can_access_project`. Currently I don't think we make the distinction clear:

- The docstring for `can_access_project` is: "Checks if the project is accessible by the user"; 
- The docstring for `can_view_project_files` is: "Checks if the project files are accessible by the user".